jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -469,12 +481,73 @@ def current_datum(self) -> dict:
             all_datums_received=True,
         )
 
-    def _get_fileinfo_path(self, fileinfo: pfs.FileInfo) -> Path:
+    def download(self):
+        """
+        Downloads the currently mounted datum to a local cache directory,
+        then links the downloaded files to PFS_MOUNT_DIR (/pfs by default).
+        Subsequent calls to this method will remove the previously
+        downloaded datum and link the newly downloaded ones.
+        """
+        if (
+            len(self._datum_list) == 0
+            or len(self._datum_list[self._datum_index].data) == 0
+        ):
+            raise ValueError("Attempting to download empty or unmounted datum")
+
+        download_dir = (
+            f"{self._DOWNLOAD_DIR}/datum-{datetime.datetime.now().isoformat()}"
+        )
+        try:
+            os.makedirs(download_dir, exist_ok=True)
+            for fileinfo in self._datum_list[self._datum_index].data:
+                path = self._get_download_path(
+                    download_dir=download_dir, fileinfo=fileinfo
+                )
+                os.makedirs(path.parent, exist_ok=True)
+                if fileinfo.file_type == pfs.FileType.FILE:
+                    # download individual file
+                    with self._client.pfs.pfs_file(file=fileinfo.file) as datum_file:
+                        with open(path, "wb") as download_file:
+                            download_file.write(datum_file.readall())
```

Comment:
Oh also, I think using `shutil.copyfileobj` might be more performant here for larger files.
https://docs.python.org/3/library/shutil.html#shutil.copyfileobj